Dịch "anchor_vn" - Phần 1 (#2256)
aivivn/d2l-vn: chapter_computer-vision/anchor_vn.md

# Anchor Boxes
-->

> # Các khung neo

Reviewer: ```suggestion
# Khung neo
```

We will practice object detection based on anchor boxes in the following sections.
-->

Các giải thuật phát hiện vật thể thường lấy mẫu ở rất nhiều vùng của ảnh đầu vào, rồi xác định các vùng đó có chứa đối tượng cần quan tâm hay không, và điều chỉnh các biên của vùng lấy mẫu này để dự đoán một cách chính xác các khung chứa chuẩn của đối tượng.
Các mô hình khác nhau có thể dùng các phương pháp lấy mẫu vùng ảnh khác nhau.

> Ở đây, chúng tôi sẽ giới thiệu một phương pháp như vậy: nó tạo ra nhiều khung chứa với kích thước và tỉ lệ cạnh khác nhau với tâm trên tất cả các điểm của ảnh .

Reviewer: ```suggestion
Ở đây, chúng tôi sẽ giới thiệu một phương pháp như vậy: nó tạo ra nhiều khung chứa với kích thước và tỉ lệ cạnh khác nhau với tâm trên mọi điểm ảnh.
```

Các khung chứa đó được gọi là các khung neo.
Chúng ta sẽ thực hành phát hiện vật thể dựa trên các khung neo trong các phần sau đây.

<!--
First, import the packages or modules required for this section.
Here, we have modified the printing accuracy of NumPy.
Because printing tensors actually calls the print function of NumPy, the floating-point numbers in tensors printed in this section are more concise.
-->

> Trước tiên, thực hiện nhập các gói và mô-đun cần thiết cho mục này.

Reviewer: ```suggestion
Trước tiên, hãy nhập các gói và mô-đun cần thiết cho mục này.
```

> Tại đây, ta đã chỉnh sửa độ chính xác in ra của Numpy.

Reviewer: ```suggestion
Tại đây, ta đã chỉnh sửa độ chính xác khi in số thực của Numpy.
```

> Do các tensor in ra thực chất gọi hàm in của Numpy, nên các số thực dấu chấm động trong trong các tensor in ra trong phần này chính xác hơn.

Reviewer: ```suggestion
Do thực chất ta đang gọi hàm in của Numpy khi in các tensor, trong phần này các số thực dấu phẩy động trong tensor được in ra sẽ chính xác hơn.
```

npx.set_np()
```

<!--
## Generating Multiple Anchor Boxes
-->

> ## Thực hiện sinh ra nhóm khung neo

Reviewer: ```suggestion
## Sinh nhiều khung neo
```

When the center position is given, an anchor box with known width and height is determined.
-->

> Giả sử ảnh đầu vào có độ cao $h$ và độ rộng $w$.

Reviewer: ```suggestion
Giả sử ảnh đầu vào có chiều cao $h$ và chiều rộng $w$.
```

> Ta sinh ra các khung neo với kích thước khác nhau với tâm tại tất cả các điểm của ảnh này.

Reviewer: ```suggestion
Ta sinh ra các khung neo với kích thước khác nhau với tâm tại mọi điểm của ảnh.
```

> Giả sử kích thước này là $s\in (0, 1]$, tỉ lệ cạnh là $r >0$, tương ứng độ cao và độ rộng của hộp neo lần lượt là $ws\sqrt{r}$ and $hs/\sqrt{r}$.

Reviewer: ```suggestion
Giả sử kích thước này $s\in (0, 1]$, tỉ lệ cạnh là $r >0$, chiều rộng và chiều cao của khung neo lần lượt là $ws\sqrt{r}$ and $hs/\sqrt{r}$.
```

> Với một vị trí tâm cho trước, ta xác định được khung neo với độ cao và độ rộng đã biết.

Reviewer: ```suggestion
Với một vị trí tâm cho trước, ta xác định được khung neo với chiều cao và chiều rộng như trên.
```

Therefore, we are usually only interested in a combination containing $s_1$ or $r_1$ sizes and aspect ratios, that is:
-->

Dưới đây ta thiết lập một tập kích thước $s_1,\ldots, s_n$ và một tập tỉ lệ khung $r_1,\ldots, r_m$.

> Nếu ta dùng tổ hợp tất cả các kích thước và tỉ lệ khung với mỗi điểm ảnh làm một tâm, ảnh đầu vào sẽ có tổng cộng  $whnm$ khung neo.

Reviewer: ```suggestion
Nếu ta dùng tổ hợp tất cả các kích thước và tỉ lệ khung với mỗi điểm ảnh làm một tâm, ảnh đầu vào sẽ có tổng cộng $whnm$ khung neo.
```

> Mặc dù các khung neo có thể bao gồm tất các khung chứa chuẩn đối tượng, độ phức tạp tính toán này thường quá nhiều.

Reviewer: ```suggestion
Mặc dù các khung chứa chuẩn đối tượng có thể sẽ nằm trong số đó, độ phức tạp tính toán này thường quá cao.
```

> Do đó, ta thường chỉ chú ý tới một tổ hợp chứa kích thước $s_1$ hoặc tỉ lệ khung $r_1$, đó là:

Reviewer: ```suggestion
Do đó, ta thường chỉ chú ý tới tổ hợp chứa $s_1$ kích thước hoặc $r_1$ tỉ lệ khung như sau:
```

* Đoàn Võ Duy Thanh
<!-- Phần 1 -->

> * Nguyễn Mai Hoàng Long

Reviewer 2: ```suggestion
* Nguyễn Mai Hoàng Long
* Phạm Minh Đức
```
